src/integrationTest/java/uk/gov/hmcts/reform/wataskmanagementapi/utils/TaskMandatoryFieldsValidatorIntegrationTest.java

```diff
@@ -0,0 +1,179 @@
+    void given_null_mandatory_field_when_validate_then_throw_service_mandatory_field_validation_exception() {
+        TaskResource task = getTaskResource(taskId);
+        task.setCaseName(null);
+        ServiceMandatoryFieldValidationException exception =
+            assertThrows(ServiceMandatoryFieldValidationException.class, ()
+                -> taskMandatoryFieldsValidator.validate(task));
+        String message = exception.getMessage();
+        assertTrue(message.contains("caseName cannot be null or empty"));
+    }
+
+    @Test
+    @DisplayName("should throw ServiceMandatoryFieldValidationException when multiple mandatory fields are missing")
+    void given_null_multiple_mandatory_field_when_validate_then_throw_service_mandatory_field_validation_exception() {
+        TaskResource task = getTaskResource(taskId);
+        task.setCaseName(null);
+        task.setCaseId("");
+        ServiceMandatoryFieldValidationException exception =
+            assertThrows(ServiceMandatoryFieldValidationException.class, ()
+                -> taskMandatoryFieldsValidator.validate(task));
+
+        String message = exception.getMessage();
+        assertTrue(message.contains("caseId cannot be null or empty"));
+        assertTrue(message.contains("caseName cannot be null or empty"));
+    }
+
+    @Test
+    @DisplayName("should throw IllegalArgumentException when property value cannot be found")
+    void given_invalid_property_when_validate_then_throw_illegal_argument_exception() {
+        TaskResource task = getTaskResource(taskId);
+        TaskMandatoryFieldsValidator validator = new TaskMandatoryFieldsValidator(
+            new LaunchDarklyFeatureFlagProvider(ldClient), true, List.of("field1", "field2"),
+            jsonParserUtils);
+        assertThrows(IllegalArgumentException.class, () -> validator.validate(task));
+    }
+
+    @Test
+    @DisplayName("should not validate when mandatory field check is disabled")
+    void given_mandatory_field_check_disabled_when_validate_then_no_validation() {
+        TaskResource task = getTaskResource(taskId);
+        TaskMandatoryFieldsValidator validator =
+            new TaskMandatoryFieldsValidator(null, false,
+                                             List.of("field1", "field2"), jsonParserUtils);
+        assertDoesNotThrow(() -> validator.validate(task));
+    }
+
+    @Test
+    @DisplayName("should not validate when jurisdiction is excluded")
+    void given_jurisdiction_excluded_then_no_validation() {
+        TaskResource task = getTaskResource(taskId);
+        task.setCaseId(null);
+        task.setJurisdiction("WA");
+
+        TaskMandatoryFieldsValidator spyValidator = Mockito.spy(taskMandatoryFieldsValidator);
+        spyValidator.validate(task);
+
+        Mockito.verify(spyValidator, Mockito.never()).validateTaskMandatoryFields(task);
+    }
+
+    @Test
+    @DisplayName("should throw ValidationException when mandatory field is missing")
```

Comment:
Is this a duplicate of given_empty_mandatory_field_when_validate_then_throw_service_mandatory_field_validation_exception above?